app/models/customer.rb

```diff
@@ -45,4 +47,22 @@ def check_for_orders
     errors[:base] << I18n.t('admin.customers.destroy.has_associated_orders')
     false
   end
+
+  # rubocop:disable Metrics/AbcSize
+  def setup_intents
```

Comment:
this is very hard to understand. I think we need to have a better name for the method and a shorter method.
Do we really need this to be a model validation? Can we move this to a service? I'd prefer to keep the customer model simple.